src/geo/ui/legends/bubble/legend-view.js

```diff
@@ -69,10 +69,23 @@ var BubbleLegendView = LegendViewBase.extend({
   },
 
   _calculateAverageSize: function () {
+    var sizes = this._calculateBubbleSizes().reverse();
```

Comment:
This algorithm is kind of hard to understand right now (at least for me)... I'd appreciate if you could clarify it a bit (adding some comments, adding some tests..., up to you!) so that anyone who will have to get in here in the future will be able to change it safely.